llama_stack/providers/tests/inference/conftest.py

```diff
@@ -78,3 +84,24 @@ def pytest_generate_tests(metafunc):
         ):
             fixtures = [stack.values[0]["inference"] for stack in filtered_stacks]
         metafunc.parametrize("inference_stack", fixtures, indirect=True)
+
+    if "embedding_model" in metafunc.fixturenames:
+        model = metafunc.config.getoption("--embedding-model")
+        if not model:
+            raise ValueError(
+                "No embedding model specified. Please provide a valid embedding model."
+            )
+        params = [pytest.param(model, id="")]
+
+        metafunc.parametrize("embedding_model", params, indirect=True)
+
+    if "embedding_stack" in metafunc.fixturenames:
```

Comment:
why do we need a separate `embedding_stack` fixture? why can't `inference_stack` work like in other tests?

Reply:
The main reason is because of embedding_model needing different registration arguments regarding the embedding_dimension compared to an inference model. But its not needed. I just need a way to know if a model is embedding model or not and if so, what the dimension is.  One simple way is to check if EMBEDDING_DIMENSION is set and if so, use the embedding model during registration. Actually, this is simpler. let me do it.